scripts/pin_the_slow_way.py

```diff
@@ -25,33 +25,80 @@
 import conda_smithy.feedstocks as feedstocks
 
 
-pinned = {
+# Some links that point to backward compatibility
+# https://abi-laboratory.pro/tracker/timeline/ffmpeg/
+# https://abi-laboratory.pro/tracker/timeline/libtiff/
+# https://abi-laboratory.pro/tracker/timeline/libzip/
+# https://abi-laboratory.pro/tracker/timeline/fontconfig/
+# https://abi-laboratory.pro/tracker/timeline/freetype/
+# https://abi-laboratory.pro/tracker/timeline/openssl/
+# https://abi-laboratory.pro/tracker/timeline/sqlite/
+# http://upstream.rosalinux.ru/versions/libgeos_c.html
+# http://upstream.rosalinux.ru/versions/libgeos.html
+# http://upstream.rosalinux.ru/versions/netcdf.html
+# http://upstream.rosalinux.ru/versions/readline.html
+# http://upstream.rosalinux.ru/versions/tk.html
+# http://upstream.rosalinux.ru/versions/libsox.html
+# http://upstream.rosalinux.ru/versions/zlib.html
+# http://upstream.rosalinux.ru/versions/ncurses.html
+# http://upstream.rosalinux.ru/versions/blitz.html
+
+build_pin = {
           'boost': 'boost 1.61.*',
           'bzip2': 'bzip2 1.0.*',
           'ffmpeg': 'ffmpeg 2.8.*',
           'fontconfig': 'fontconfig 2.11.*',
           'freetype': 'freetype 2.6.*',
           'geos': 'geos 3.4.*',
-          'giflib': 'giflib 5.1.*',
+          'giflib': 'giflib 5.1.2',
           'hdf5': 'hdf5 1.8.17|1.8.17.*',
           'icu': 'icu 56.*',
           'jpeg': 'jpeg 9*',
           'libblitz': 'libblitz 0.10|0.10.*',
           'libmatio': 'libmatio 1.5.*',
-          'libnetcdf': 'libnetcdf 4.4.*',
-          'libpng': 'libpng >=1.6.21,<1.7',
+          'libnetcdf': 'libnetcdf 4.2.*',
+          'libpng': 'libpng 1.6.21',
           'libsvm': 'libsvm 3.21|3.21.*',
           'libtiff': 'libtiff 4.0.*',
           'ncurses': 'ncurses 5.9*',
           'openblas': 'openblas 0.2.18|0.2.18.*',
-          'openssl': 'openssl 1.0.*',
+          'openssl': 'openssl 1.0.2g',
           'readline': 'readline 6.2*',
           'sox': 'sox 14.4.2',
-          'sqlite': 'sqlite 3.13.*',
+          'sqlite': 'sqlite 3.7.*',
           'tk': 'tk 8.5.*',
           'vlfeat': 'vlfeat 0.9.20',
           'xz': 'xz 5.2.*',
-          'zlib': 'zlib 1.2.*',
+          'zlib': 'zlib 1.2.7',
+        }
+
+run_pin = {
+          'boost': 'boost 1.61.*',
+          'bzip2': 'bzip2 >=1.0,<2',
+          'ffmpeg': 'ffmpeg >=2.8,<3.0',
```

Comment:
According to the reports, they tend to break compatibility (medium severity) between minors. So I think it should be really ```2.8.*```